File: cslavb/Csla/Validation/IAsyncRuleMethod.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Validation
  Friend Interface IAsyncRuleMethod
    Inherits IRuleMethod
    ReadOnly Property AsyncRuleArgs() As AsyncRuleArgs
    ReadOnly Property Severity() As RuleSeverity
    Overloads Sub Invoke(ByVal target As Object, ByVal complete As AsyncRuleCompleteHandler)
  End Interface
End Namespace

File: cslacs/Csla/Validation/IAsyncRuleMethod.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Csla.Validation
{
  internal interface IAsyncRuleMethod : IRuleMethod
  {
    AsyncRuleArgs AsyncRuleArgs { get; }

    RuleSeverity Severity { get; }

    void Invoke(object target, AsyncRuleCompleteHandler complete);
  }
}
